library/kani_core/src/lib.rs

```diff
@@ -360,6 +365,13 @@ macro_rules! kani_intrinsics {
             #[doc(hidden)]
             #[rustc_diagnostic_item = "KaniInitContracts"]
             pub fn init_contracts() {}
+
+            /// This should only be used within contracts. The intent is to
+            /// perform type inference on a closure's argument
+            #[doc(hidden)]
+            pub fn apply_closure<T, U: Fn(&T) -> bool>(f: U, x: &T) -> bool {
```

Comment:
It's not clear to me why we're not removing `apply_closure` in `library/kani/src/internal.rs`.

Reply:
Thanks for reminding me. We should be removing the entire file `internal.rs` itself, as it is simply duplicated code.

Reply:
Redundant file has been removed.